### ISSUE TITLE
[Vulkan][Mali] BERT encoder loses the device and Whisper encoder returns all-NaN on Mali-G76

### ISSUE THREAD

**Issue by msluszniak** [pytorch/executorch#22322]
### 🐛 Describe the bug

Two independent failures of the Vulkan backend's transformer path on a Mali-G76 (Galaxy S10+, SM-G975F, Android 12, driver `v1.r32p1`). Both are 100% reproducible. Neither is root-caused; filing because they are cheap to reproduce and the same models are fine on an Adreno 840.

Everything below is ExecuTorch `main` @ c27baa8031, arm64-v8a, NDK 27.1, `executor_runner` built with `-DEXECUTORCH_BUILD_VULKAN=ON -DEXECUTORCH_ENABLE_LOGGING=ON`.

#### 1. BERT encoder hangs the GPU (`VK_ERROR_DEVICE_LOST`)

`sentence-transformers/all-MiniLM-L6-v2` (6-layer BERT, 128 tokens) lowers to 2 delegate blobs and then loses the device on first execute:

```
I executorch:executor_runner.cpp:634] Model loaded in 2600.294613 ms.
libc++abi: terminating due to uncaught exception of type vkcompute::vkapi::Error:
Exception raised from wait at backends/vulkan/runtime/vk_api/Fence.cpp:66:
(fence_status != VK_ERROR_DEVICE_LOST) is false! Vulkan Fence: Device lost while waiting for fence!
```

This happens both **with** and **without** an attention mask (the mask-free variant is a single delegate blob), so it is not the mask path. Note the same model runs correctly on an Adreno 840 in 7.8 ms, so it is not a timeout on absolute work: the whisper encoder below takes 1.5 s per iteration on this same Mali device without losing the device.

#### 2. Whisper encoder returns all-NaN

`openai/whisper-tiny`'s encoder (mel input `[1, 80, 3000]`, one delegate blob) runs to completion and returns **NaN for all 576000 output elements**, identically across 12 in-process replays:

```
whisper MALI vulkan output: 576000 elems, NaN=576000 (100.0%), Inf=0
```

The same `.pte` on an Adreno 840 returns zero NaNs. XNNPACK on this same Mali device matches the eager reference.

#### Control: the conv-only model is fine

`selfie_segmentation` @256 (54 conv2d, one delegate blob) is **bit-exact** against the eager reference on this device (max abs deviation 1.19e-07) and stable across 60 in-process replays. So the Vulkan backend is working on this GPU in general; the failures are specific to the transformer models.

### Reproducing

Export either model with `VulkanPartitioner` (use `attn_implementation="eager"` for a clean partition) and run:

```
./executor_runner --model_path <model>_vulkan.pte --inputs <in>.bin \
    --num_executions 12 --print_output=none
```

I am happy to run further experiments on this device, including any instrumented build. Bisecting which op produces the NaN would need guidance on the least painful way to dump intermediate delegate tensors.

### Versions

ExecuTorch main @ c27baa8031. Galaxy S10+ (SM-G975F), Mali-G76, Android 12. Contrast device: Galaxy S26 Ultra, Adreno 840, where both models run without these failures.

cc @SS-JIA @manuelcandales @digantdesai @cbilgin

**Comment by msluszniak** [comment]
closing in favour of #22323
